tests/Unit/Infrastructure/Event/TwigGlobalsListenerTest.php

```diff
@@ -0,0 +1,121 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * Copyright (c) 2013-2017 OpenCFP
+ *
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ *
+ * @see https://github.com/opencfp/opencfp
+ */
+
+namespace OpenCFP\Test\Unit\Infrastructure\Event;
+
+use Mockery;
+use OpenCFP\Domain\CallForPapers;
+use OpenCFP\Domain\Services\Authentication;
+use OpenCFP\Infrastructure\Auth\UserInterface;
+use OpenCFP\Infrastructure\Event\TwigGlobalsListener;
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\EventDispatcher\EventDispatcher;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Session\Session;
+use Symfony\Component\HttpFoundation\Session\Storage\MockArraySessionStorage;
+use Symfony\Component\HttpKernel\Event\GetResponseEvent;
+use Symfony\Component\HttpKernel\HttpKernelInterface;
+use Symfony\Component\HttpKernel\KernelEvents;
+use Twig_Environment;
+
+/**
+ * @covers \OpenCFP\Infrastructure\Event\TwigGlobalsListener
+ */
+final class TwigGlobalsListenerTest extends TestCase
+{
+    /**
+     * @dataProvider provideTestSetup
+     */
+    public function testGlobals(Authentication $authentication, bool $isOpen, string $uri, string $flash = null, string $fixture)
+    {
+        $twig    = new Twig_Environment(new \Twig_Loader_Filesystem(__DIR__ . '/Fixtures'));
+        $session = new Session(new MockArraySessionStorage());
+
+        if ($flash !== null) {
+            $session->set('flash', $flash);
+        }
+
+        $eventDispatcher = new EventDispatcher();
+        $eventDispatcher->addSubscriber(new TwigGlobalsListener(
+            $authentication,
+            $this->mockCallForPapers($isOpen),
+            $session,
+            $twig
+        ));
+
+        $eventDispatcher->dispatch(KernelEvents::REQUEST, new GetResponseEvent(
+            Mockery::mock(HttpKernelInterface::class),
+            Request::create($uri),
+            HttpKernelInterface::MASTER_REQUEST
+        ));
+
+        $output = $twig->render('globals.txt.twig');
+
+        $this->assertStringEqualsFile(__DIR__ . '/Fixtures/' . $fixture, $output);
+    }
+
+    public function provideTestSetup(): iterable
```

Comment:
```
Fatal error: Generators may only declare a return type of Generator, Iterator or Traversable, OpenCFP\Test\Unit\Infrastructure\Event\iterable is not permitted in /home/travis/build/opencfp/opencfp/tests/Unit/Infrastructure/Event/TwigGlobalsListenerTest.php on line 69
```

This fails on PHP 7 because `iterable` wasn't a thing at that point. 

Could this `@dataProvider` be modified (and return type adjusted):

```
return [
    [$this->mockAdmin(), true, '/foo', null, 'global_with_admin.txt'],
    [$this->mockReviewer(), true, '/foo', 'You\'ve got mail.', 'global_with_reviewer.txt'],
    [$this->mockAnonymous(), false, '/bar', null, 'global_with_anonymous_user.txt']
]
```

I'm unsure what benefit this provides beyond personal preference. I'm approaching this mostly from a consistency perspective. Obviously, if there is clear reason to return an `iterable` for a specific `TestCase`, we can do it. The examples I've seen recently coming in with contributions seem to be preference-based.

Reply:
Fixed.